### PR TITLE
Latest updates for LabelLayer

### PR DESCRIPTION
**Pull request by ibgreen**

### CODE REVIEW

File: examples/sample-layers/label-layer/label-utils.js [hunk 1]
```diff
@@ -21,48 +15,37 @@ function setTextStyle(ctx, fontSize) {
...
       object,
       index,
       string,
-      x,
-      y: row * fontSize,
-      width: Math.min(width, MAX_CANVAS_WIDTH),
-      height: fontSize,
-      mask: true
+      mask: true,
+      ...region,
```
Comment (heshan0131) [inline examples/sample-layers/label-layer/label-utils.js]: Object.assign(...)

File: examples/sample-layers/label-layer/label-layer.js [hunk 2]
```diff
@@ -4,8 +4,10 @@ import {makeTextureAtlasFromLabels} from './label-utils';
 /* Constants */
 const defaultProps = {
   getLabel: x => x.label,
-  fontSize: 24,
-  fp64: false
+  getPosition: x => x.position,
+  getSize: x => 1,
+  getColor: x => [0, 0, 0, 255],
+  fontSize: 24
 };
```
Comment (shaojingli) [inline examples/sample-layers/label-layer/label-layer.js]: fp64 flag removed?

File: examples/sample-layers/label-layer/label-utils.js [hunk 3]
```diff
@@ -21,48 +15,37 @@ function setTextStyle(ctx, fontSize) {
 export function makeTextureAtlasFromLabels(gl, {data, getLabel = x => x, fontSize = 48}) {
   const canvas = document.createElement('canvas');
```
Comment (shaojingli) [inline examples/sample-layers/label-layer/label-utils.js]: So we are creating a new canvas + 2d context every time text is changed. Is there a need to cache the canvas?

File: examples/sample-layers/label-layer/label-layer.js (hunk 2 above)
Comment (Pessimistress) [inline examples/sample-layers/label-layer/label-layer.js]: Should be able to just pass `fp64` prop to the IconLayer

File: examples/sample-layers/label-layer/label-layer.js [hunk 4]
```diff
@@ -37,12 +41,14 @@ export default class LabelLayer extends CompositeLayer {
...
   renderLayers() {
+    const {fontSize} = this.props;
+
     return new IconLayer(Object.assign({}, this.props, {
       id: 'label-icons',
       iconAtlas: this.state.texture,
       iconMapping: this.state.mapping,
       data: this.state.mapping,
-      sizeScale: 24,
+      sizeScale: fontSize,
```
Comment (Pessimistress) [inline examples/sample-layers/label-layer/label-layer.js]: If we're still using the `fontSize`, should keep it in `defaultProps` with value: 1

File: examples/sample-layers/label-layer/label-utils.js [hunk 5]
```diff
@@ -21,48 +15,37 @@ function setTextStyle(ctx, fontSize) {
...
+      anchorY: region.height
     };
-    x += width;
     return iconMap;
   });
 
+  const extents = textPainter.getExtents();
   canvas.width = MAX_CANVAS_WIDTH;
-  canvas.height = (row + 1) * fontSize;
+  canvas.height = extents.height;
```
Comment (Pessimistress) [inline examples/sample-layers/label-layer/label-utils.js]: `TextAtlasPainter` does not force dimensions to power of two. Does this still work with luma.gl 4?

**Comment by shaojingli** [comment]
Let me close this PR as it might be no longer valid for new version of deck.gl
@rivulet-zhang will attempt to reorg our label, icon, multi-icon and text layers

**Pull request closed without merging**